connectivity/check/policy.go

```diff
@@ -454,12 +455,19 @@ func sumMap(m map[string]int) int {
 	return sum
 }
 
+// policyApplyDeleteLock guarantees that only one connectivity test instance
+// can apply or delete policies in case of connectivity test concurrency > 1
+var policyApplyDeleteLock = sync.Mutex{}
+
 // applyPolicies applies all the Test's registered network policies.
 func (t *Test) applyPolicies(ctx context.Context) error {
 	if len(t.cnps) == 0 && len(t.ccnps) == 0 && len(t.knps) == 0 && len(t.cegps) == 0 {
 		return nil
 	}
 
+	policyApplyDeleteLock.Lock()
```

Comment:
I guess nothing stops a test running with CCNP from deleting it and causing other test to fail if there is a name collision between CCNPs.

Why do we need this lock in the first place? Every test is only managing it's own policies isn't it?

Unless it's about policy revision numbers, which I guess are racey.

Reply:
yeah it is about policy revision numbers unfortunately 😥 

so right now the connectivity test waits for the revision number to get incremented after applying / deleting policies. the idea is to wait until cilium reconciles the policy before generating traffic.

i vaguely remember chatting with @joestringer about this, i wanted a more reliable way to check whether cilium-agent implemented a particular CNP revision. not sure if anything concrete came out of it 💭

Reply:
Right, there's no great mechanism to create a CNP and then find out whether it's been implemented or not. The policy revision number stuff helps in a sense, because each time you create a policy, it changes this number. But there's no way to correlate the revision number with the specific policy, so we mainly rely on being the only one to create policy during a window, then if the number increases, we know it was implemented.

Typically this is something you might expect to show up in a status field for a CR. Unfortunately since policy implementation is per-node, this ends up with `N` policies each reporting the implementation status for `M` nodes, which scales poorly. IIRC we even removed the option to even report CNP status recently (https://github.com/cilium/cilium/pull/29395).

This would be a good topic to bring up with @cilium/sig-policy (#sig-policy on Slack) to discuss what reasonable options there may be to support this workflow. I think it'd be quite useful to users as well if we came up with a good solution to this problem.

Reply:
Yeah, the lock has been added due to the policy revision number.
Is it ok to resolve this conversation?